src/main/java/meilisearch/MeilisearchHttpRequest.java

```diff
@@ -0,0 +1,111 @@
+package meilisearch;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+
+class MeilisearchHttpRequest {
+	private final MeilisearchConfig config;
+
+	protected MeilisearchHttpRequest(MeilisearchConfig config) {
+		this.config = config;
+	}
+
+	/**
+	 * Create and get a validated HTTP connection to url with method and API key
+	 *
+	 * @param url    URL to connect to
+	 * @param method HTTP method to use for the connection
+	 * @param apiKey API Key to use for the connection
+	 * @return Validated connection (otherwise, will throw a {@link IOException})
+	 * @throws IOException If unable to establish connection
+	 */
+	private HttpURLConnection getConnection(final URL url, final String method, final String apiKey) throws IOException {
+		if (url == null || "".equals(method)) throw new IOException("Unable to open an HttpURLConnection with no URL or method");
+
+		HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+		connection.setRequestMethod(method);
+		connection.setRequestProperty("Content-Type", "application/json");
+
+		// Use API key header only if one is provided
+		if (!"".equals(apiKey)) {
```

Comment:
Changed `apiKey.equals("")` to this to avoid potential NPE

Reply:
@AmiralBl3ndic great~